DxDispatch/src/dxdispatch/DmlDispatchable.cpp

```diff
@@ -104,87 +146,328 @@ void FillBindingData(
 
+    for (const auto& edge : serializedDesc.IntermediateEdges) 
+    {
+        const auto& nodeDesc = serializedDesc.Nodes[edge.FromNodeIndex].Desc;
+        if (std::holds_alternative<DmlSerializedGraphNodeConstantVariant>(nodeDesc)) 
         {
-            dmlGraphNodeDesc.Type = DML_GRAPH_NODE_TYPE_OPERATOR;
-            dmlGraphNodeDesc.Desc =  &nodeDesc;
+            const auto& constantVariant = std::get<DmlSerializedGraphNodeConstantVariant>(nodeDesc);
+            if (std::holds_alternative<ConstantName>(constantVariant)) 
+            {
+                constantNodeEdges[edge.FromNodeIndex] = edge;
+            }
         }
+    }
 
-        dmlInputGraphEdges.resize(m_desc.bindPoints.inputs.size());
-        for( size_t i = 0; i < m_desc.bindPoints.inputs.size(); i++)
+        for (uint32_t nodeIndex = 0; nodeIndex < serializedDesc.Nodes.size(); nodeIndex++) 
         {
-            if (m_desc.bindPoints.inputs[i].requiredBinding)
+            const auto& node = serializedDesc.Nodes[nodeIndex];
+            if (std::holds_alternative<DmlSerializedGraphNodeConstantVariant>(node.Desc)) 
             {
-                DML_INPUT_GRAPH_EDGE_DESC desc = {};
-                desc.GraphInputIndex = gsl::narrow_cast<UINT>(i);
-                desc.ToNodeIndex = 0;
-                desc.ToNodeInputIndex = gsl::narrow_cast<UINT>(i);
-                desc.Name = m_desc.bindPoints.inputs[i].name.c_str();
-                dmlInputGraphEdges[i] = desc;
-                dmlInputEdges.push_back({ DML_GRAPH_EDGE_TYPE_INPUT, &dmlInputGraphEdges[i] });
+                const auto& constantVariant = std::get<DmlSerializedGraphNodeConstantVariant>(node.Desc);
+                if (std::holds_alternative<ConstantName>(constantVariant) && constantNodeEdges.find(nodeIndex) != constantNodeEdges.end()) 
+                {
+                    const auto& edge = constantNodeEdges[nodeIndex];
+                    const auto& operatorNode = serializedDesc.Nodes[edge.ToNodeIndex];
+                    if (std::holds_alternative<AbstractOperatorDesc>(operatorNode.Desc)) 
+                    {
+                        const auto& opDesc = std::get<AbstractOperatorDesc>(operatorNode.Desc);
+                        auto inputTensors = opDesc.GetInputTensors();
+                        if (edge.ToNodeInputIndex < inputTensors.size()) 
+                        {
+                            const auto& constantTensor = inputTensors[edge.ToNodeInputIndex];
+                            constantDataTypes[node.Name] = constantTensor->dataType;
+                        }
+                    }
+                }
             }
         }
 
-        dmlOutputGraphEdges.resize(m_desc.bindPoints.outputs.size());
-        for( size_t i = 0; i < m_desc.bindPoints.outputs.size(); i++)
+        return constantDataTypes;
+}
+
+Dispatchable::Bindings GenerateInitialBindingsFromGraph(const DmlSerializedGraphDesc& serializedDesc,
```

Comment:
As a good coding style we either keep all parameters in same line or in a new line. If you decide to move each parameter into a new line, then please indent those. Right now indentation of line 240 is off by 1 tab.